### ISSUE TITLE
Debugging - Understanding the Differences between the freeCodeCamp and Browser Console / Firefox bug

### ISSUE THREAD

**Issue by Chalarangelo** [freeCodeCamp/freeCodeCamp#17387]
#### Describe your problem and - if possible - how to reproduce it
In the JS course, the **Debugging - Understanding the Differences between the freeCodeCamp and Browser Console** lesson doesn't properly evaluate on Firefox, showing `invalid regexp group` in the bottom console. On Chrome, however, it works just fine. The issue has to do with the regexp used to check `console.clear()`.

#### Add a Link to the page with the problem

https://learn.freecodecamp.org/javascript-algorithms-and-data-structures/debugging/understanding-the-differences-between-the-freecodecamp-and-browser-console

#### Tell us about your browser and operating system

* Browser Name: Firefox
* Browser Version: Developer Edition / 61.0b10 (64-bit)
* Operating System: Windows 10

#### If possible, add a screenshot here

<img width="960" alt="fcc" src="https://user-images.githubusercontent.com/8281875/40872418-88da1b3a-6656-11e8-998c-38a396161726.PNG">

**Comment by igalbn** [comment]
The same problem. On Linux Mint 18.3 Firefox 60.0.1.

**Comment by LTKills** [comment]
Also had this problem on Debian 9.4 Firefox 60.0.1 (64-bit)

**Comment by brimarq** [comment]
Same here. MacOS 10.12.6 Firefox 60.0.1 (64-bit)

**Comment by sinansonmez** [comment]
Same here Windows Firefox

**Comment by matthew-t-smith** [comment]
This may be an issue with the validity of using the negative look behind in the regexp in the test. I have no idea how to test whether they should be valid or implementable though.

**Comment by BrittanyAN** [comment]
Same problem. Windows 10 Home 1803 / 17134.112 and Firefox 60.0.2

**Comment by scissorsneedfoodtoo** [comment]
@matthew-t-smith, I think you're right. I put the regex from the text string into regexer to test and got a warning that the negative look behind isn't supported in all browsers. It's not the most refined thing, but maybe the test could something like:

`/^(\\s*console.clear\\(\\)\\s*)$/gm`

That should pick up just the console.clear() from the code the user enters, rather than the one that's commented out. Should also handle any white space that might be in the user's code, too.

**Comment by SoyDiego** [comment]
Same problem here. Firefox Quantum 60.0.2 (64-bits)

**Comment by AymenAlSalem** [comment]
``// running test
invalid regexp group
// tests completed``
FireFox Quantum Developer Edition 62.0b1 (64-bit) win10.

**Comment by glushkov89** [comment]
FireFox Quantum 60.0.2 64bit win10

**Comment by lynxlynxlynx** [comment]
This got added to the standard only this year and yes, firefox hasn't implemented it yet:
https://bugzilla.mozilla.org/show_bug.cgi?id=1225665

**Comment by matthew-t-smith** [comment]
Well, I've got a PR that will achieve the same effect for the test that fails without the look behind. As soon as it gets approved and merged into production, we can all pass this challenge. Unfortunately, there isn't a workaround solution because it's the test that fails, not the solutions.

**Comment by Stesic** [comment]
_**Solution**_
// Open your browser console
let outputTwo = "This will print to the browser console 2 times";
// Use console.log() to print the outputTwo variable
console.log(outputTwo);

let outputOne = "Try to get this to log only once to the browser console";
// Use console.clear() in the next line to print the outputOne only once
console.clear();

// Use console.log() to print the outputOne variable
console.log(outputOne);

**Comment by vytautas-pilk** [comment]
Same here Linux SolusOS, Firefox 60.0.1

**Comment by Brown1ee** [comment]
This code help me to pass the lesson
![capture-20180827-095639](https://user-images.githubusercontent.com/42550136/44645239-a4a6dc00-a9df-11e8-9c40-aadc52373ff6.jpg)

**Comment by scissorsneedfoodtoo** [comment]
@Brown1ee, your code looks correct. Can I confirm that you were able to pass the lesson?

This was fixed way back in PR [#56](https://github.com/freeCodeCamp/curriculum/pull/56) over in the Curriculum repo. I'll close this issue, but please feel free to comment if anyone is still having problems with this challenge.

**Comment by RobCM** [comment]
This worked for me:
// Open your browser console
let outputTwo = "This will print to the browser console 2 times";
console.log(outputTwo); //to print the outputTwo variable
let outputOne = "Try to get this to log only once to the browser console";
console.clear(); //in the next line to print the outputOne only once
console.clear();
console.log(outputOne) //to print the outputOne variable

**Comment by matthew-t-smith** [comment]
Glad to hear it worked. We got a fix implemented, so it should be all good now 👍

**Comment by sirpepo** [comment]
Mac OS with google chrome!!! Not working. Any of suggested code works for me.
I will try it on my linux mint pc
<img width="1280" alt="screen shot 2018-09-30 at 13 39 57" src="https://user-images.githubusercontent.com/9752135/46257213-5beab300-c4b6-11e8-8380-c09c37b90ca1.png">

**Comment by sirpepo** [comment]
> Mac OS with google chrome!!! Not working. Any of suggested code works for me.
> I will try it on my linux mint pc
> <img alt="screen shot 2018-09-30 at 13 39 57" width="1280" src="https://user-images.githubusercontent.com/9752135/46257213-5beab300-c4b6-11e8-8380-c09c37b90ca1.png">

OK, I fixed my own code to work, but I think you may review the tests.
I always add space between parenthesis and the content on the parenthesis for best reading, and while the code works, the tests not pass with this spaces. I'll be aware with this in next lessons.

Thanks,

**Comment by tb47** [comment]
Console outputs:

Konsole wurde geleert. understanding-the-differences-between-the-freecodecamp-and-browser-console:28:1
Try to get this to log only once to the browser console frame.js:90:11 

I go on learning ... hopeful I don`t miss too much...
